charmcraft/commands/store/client.py

```diff
@@ -44,11 +45,42 @@
 STORAGE_BASE_URL = 'https://storage.staging.snapcraftcontent.com'
 
 
+def _is_ci_env() -> bool:
+    """Utilize standard env variables to determine if the environment is for testing."""
+    env_prefixes = ["TRAVIS", "AUTOPKGTEST_TMP"]
+    for prefix in env_prefixes:
+        for key in os.environ.keys():
+            if key.startswith(prefix):
+                return True
+    return False
+
+
+def _get_os_platform(filepath: str = "/etc/os-release") -> str:
+    """Determine a system/release combo for an OS using /etc/os-release if available."""
+    if platform.system() == "Linux":
+        os_release = dict()
+        try:
+            with open(filepath) as f:
+                for line in f:
+                    entry = line.rstrip().split("=")
+                    if len(entry) == 2:
+                        os_release[entry[0]] = entry[1].strip('"')
+        except FileNotFoundError:
+            logger.warning("Unable to locate 'os-release' file, using default values")
+        finally:
+            return "{}/{} ({})".format(os_release.get("NAME", "Unknown"),
+                                       os_release.get("VERSION_ID", "Unknown Version"),
+                                       platform.machine())
+
+    return "{}/{} ({})".format(platform.system(), platform.release(), platform.machine())
```

Comment:
Don't format the platform twice (as this could lead to inconsistencies in the future because changing only one).

So, just do `"{}/{} ({})".format(system, release, machine)` at the end, and arrive there with the proper values for each of those.